scripts/zones/QuBia_Arena/mobs/Warlord_Rojgnoj.lua

```diff
@@ -7,16 +7,28 @@ mixins = {require("scripts/mixins/job_special")}
 require("scripts/globals/status")
 local ID = require("scripts/zones/QuBia_Arena/IDs")
 
+function allHeirMobsDead(mob)
+    local battlefield = mob:getBattlefield()
+    local inst = battlefield:getArea()
+    local instOffset = ID.mob.HEIR_TO_THE_LIGHT_OFFSET + (14 * (inst-1))
+    for i = instOffset + 3, instOffset + 13 do
+        if not GetMobByID(i):isDead() then
+            return false
+        end
+    end
+    return true
+end
+
 function onMobInitialize(mob)
     mob:addMod(tpz.mod.SLEEPRES,50)
 end
 
-function onMobDeath(mob, player, isKiller)
-    local inst = mob:getBattlefield():getArea()
-    local instOffset = ID.mob.HEIR_TO_THE_LIGHT_OFFSET + (14 * (inst-1))
-    if GetMobByID(instOffset + 0):isDead() and GetMobByID(instOffset + 1):isDead()
-        and GetMobByID(instOffset + 2):isDead()
-    then
-        mob:getBattlefield():setLocalVar("loot", 0)
+function onMobSpawn(mob)
+    local battlefield = mob:getBattlefield()
+    if allHeirMobsDead(mob) then
```

Comment:
1) Maybe rename this function to "phaseOneDead" or "phaseOneComplete"

2) Can pass in the battlefield directly into the function instead of the mob; the original mob isn't used once inside it

Reply:
`phaseChangeReady()` ?

Reply:
True to both points. I just copied this function from the phase 1 mobs, should I change it there too?

Reply:
Yeah, "allHeirMobsDead" is even more confusing inside the phase 1 mob files~! 😅